### PR TITLE
callback when player is played/paused through notification area

### PR DESCRIPTION
**Pull request by subhash279**
A new callback to listen to player state change from notification area.   onNotificationPlayerStateChanged is similar to existing onPlayerStateChanged except that onNotificationPlayerStateChanged is called when a user interacts through the notification area. This is useful to change the app UI based on user interaction in the notification area.

### CODE REVIEW

File: example/lib/player_widget.dart [hunk 1]
```diff
@@ -173,6 +173,13 @@ class _PlayerWidgetState extends State<PlayerWidget> {
         _audioPlayerState = state;
       });
     });
+
+    _audioPlayer.onNotificationPlayerStateChanged.listen((state) {
+      if (!mounted) return;
+      setState(() {
+        _audioPlayerState = state;
+      });
```
Comment (luanpotter) [inline example/lib/player_widget.dart]: inline this function

File: lib/audioplayers.dart [hunk 2]
```diff
@@ -397,6 +411,14 @@ class AudioPlayer {
     final value = callArgs['value'];
 
     switch (call.method) {
+      case 'audio.onNotificationPlayerStateChanged':
+        final int isPlaying = value;
```
Comment (luanpotter) [inline lib/audioplayers.dart]: return is player as a boolean if it has only two states

File: lib/audioplayers.dart [hunk 3]
```diff
@@ -397,6 +411,14 @@ class AudioPlayer {
...
+        if (isPlaying == 1) {
+          player.notificationState = AudioPlayerState.PLAYING;
+        } else if (isPlaying == 0) {
+          player.notificationState = AudioPlayerState.PAUSED;
```
Comment (luanpotter) [inline lib/audioplayers.dart]: with a boolean you can use a ternary here

**Comment by subhash279** [comment]
@luanpotter The changes are done :)

**Comment by bardiarastin** [comment]
Dear @luanpotter your requested changes are done, is there any special reason you've not merged this yet?!

**Comment by subhash279** [comment]
@luanpotter I fixed any minor bugs on ios notification side and this is the final push for ios notifications. Can you please merge this and then I will start on the android side.

**Pull request merged**